### PR TITLE
Refactor configuration processing

### PR DESCRIPTION
**Pull request by CraigFe**
Extraction of some refactoring from https://github.com/mirage/alcotest/pull/294. Defines config keys and their defaults in one place, and avoids shunting individual flags around in the Cli definition.

### CODE REVIEW

**Review by Ngoguey42 — COMMENTED** [review]
LGTM

File: src/alcotest-engine/config.ml [hunk 1]
```diff
@@ -1,35 +1,248 @@
...
+      show_errors = merge_on (fun t -> t.show_errors);
+      json = merge_on (fun t -> t.json);
+      filter = merge_on (fun t -> t.filter);
+      log_dir = merge_on (fun t -> t.log_dir);
+      bail = merge_on (fun t -> t.bail);
+    }
+
+  let term ~and_exit =
+    let open Cmdliner_syntax in
+    let open Key in
```
[the remark this answers is not in the document]
Comment (CraigFe) [reply src/alcotest-engine/config.ml]: Really wish the compiler would warn me about this...

File: src/alcotest-engine/config_intf.ml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  include module type of Types
+
+  module User : sig
+    type t
+
+    val kcreate : (t -> 'a) -> 'a with_options
+    val create : (unit -> t) with_options
+    val term : and_exit:bool -> t Cmdliner.Term.t
+    val ( || ) : t -> t -> t
+    val and_exit : t -> bool
```
[the remark this answers is not in the document]
Comment (CraigFe) [reply src/alcotest-engine/config_intf.ml]: Have added some documentation; let me know what you think.

[the remark this answers is not in the document]
Comment (Ngoguey42) [reply src/alcotest-engine/config_intf.ml]: Great

**Pull request merged**